### PR TITLE
pull status handling

### PR DESCRIPTION
**Pull request by scottf**

### CODE REVIEW

**Review by RichardHightower — APPROVED** [review]

File: src/main/java/io/nats/client/JetStreamStatusException.java [hunk 1]
```diff
@@ -29,8 +32,13 @@ public class JetStreamStatusException extends IllegalStateException {
      * @param status the status
      */
     public JetStreamStatusException(JetStreamSubscription sub, Status status) {
-        super("Unknown or unprocessed status message: " + status.getMessage());
+        this(sub, DEFAULT_DESCRIPTION, status);
+    }
+
+    public JetStreamStatusException(JetStreamSubscription sub, String description, Status status) {
```
Comment (arondi) [inline src/main/java/io/nats/client/JetStreamStatusException.java:38]: needs javadoc

File: src/main/java/io/nats/client/impl/NatsJetStreamPullSubscription.java [hunk 2]
```diff
@@ -331,4 +333,9 @@ public void stop() {
     public JetStreamReader reader(final int batchSize, final int repullAt) {
         return new JetStreamReaderImpl(this, batchSize, repullAt);
     }
+
+    @Override
```
Comment (arondi) [inline src/main/java/io/nats/client/impl/NatsJetStreamPullSubscription.java:337]: needs inheritDoc

File: src/main/java/io/nats/client/impl/NatsJetStreamSubscription.java [hunk 3]
```diff
@@ -233,6 +234,11 @@ public ConsumerInfo getConsumerInfo() throws IOException, JetStreamApiException
         return js.lookupConsumerInfo(stream, consumerName);
     }
 
+    @Override
```
Comment (arondi) [inline src/main/java/io/nats/client/impl/NatsJetStreamSubscription.java:237]: needs inheritDoc

File: src/main/java/io/nats/client/impl/NatsSubscription.java [hunk 4]
```diff
@@ -68,11 +68,11 @@ public boolean isActive() {
         return (this.dispatcher != null || this.incoming != null);
     }
 
-    void setBeforeQueueProcessor(Function<NatsMessage, NatsMessage> beforeQueueProcessor) {
-        this.beforeQueueProcessor = beforeQueueProcessor; // better not be null if it's being set
+    void setBeforeQueueProcessor(Function<NatsMessage, Boolean> beforeQueueProcessor) {
```
Comment (arondi) [inline src/main/java/io/nats/client/impl/NatsSubscription.java:71]: does this need to be synchronized guarded?

File: src/main/java/io/nats/client/support/PullStatus.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+// limitations under the License.
+
+package io.nats.client.support;
+
+public class PullStatus {
+    private final long pendingMessages;
+    private final long pendingBytes;
+    private final boolean trackingHeartbeats;
+
+    public PullStatus(long pendingMessages, long pendingBytes, boolean trackingHeartbeats) {
```
Comment (arondi) [inline src/main/java/io/nats/client/support/PullStatus.java:21]: javadoc

**Review by arondi — CHANGES_REQUESTED** [review]

File: src/main/java/io/nats/client/impl/NatsSubscription.java (hunk 4 above)
Comment (scottf) [reply src/main/java/io/nats/client/impl/NatsSubscription.java:71]: No. This is used internally during construction or post construction

**Pull request merged**